### DISCUSSION TITLE
Getting object with all of its deep references resolved

### DISCUSSION THREAD

**Discussion by dimitarkolev** [directus/directus#12392] [Archive]
Developing a plugin that needs to get an object with all nested references. Few versions ago this worked:
```
await itemsService.readOne(key, { fields: '*.*.*' })
```
With recent version id doesn't. I don't find any documentation that is related to itemsService but only about the REST Api. I've tried few things I found there but without any success.
What is the best way go get an item with all its relations resolved?

**Comment by rijkvanzanten** [comment]
Fields is an array of strings, other than that the above should still work 👍🏻 

```js
await itemsService.readOne(key, { fields: ['*.*.*'] })
```

**Reply by dimitarkolev** [reply]
10x this works. Is there a way to change `*.*.*` and have unlimited depth?

**Reply by rijkvanzanten** [reply]
No, as that would almost guaranteed make everything go 💥 

You can easily have a circular dependency, and end up fetching literal unlimited amounts of data

**Reply by rijkvanzanten** [reply]
In general, I'd highly recommend fetching just the data you actually need, not wildcard everything as that's way less optimized 👍🏻

**Reply by dimitarkolev** [reply]
Our plugin is doing data syncing with another service, so no easy way to define what`s really needed :)

**Reply by rijkvanzanten** [reply]
Fair! In that case you might want to consider looping over all collections, and only fetching `*` for each collection. That way you prevent every fetching things circularly nested, and you're still guaranteed to have synced over all data

**Reply by dimitarkolev** [reply]
Thought about that but the service is document oriented and has no relations. We will end up implementing what deep is doing at the end. Anyway it is working now with some minor limitations which are easily put in a config file, like the default depth for loading items.
There are many solutions for circular dependency resolution. They are very common when serializing objects but maybe there are much more needed additions to directus.
